python3Packages.python-wayland-extra: init at 0.7.0, python3Packages.streamcontroller-streamdeck: init at 0.1.5, Streamcontroller: 1.5.0-beta.8 -> 1.5.0-beta.12

sifmelcara opened the pull request:
Fixes #416461

 @ciferkey 

## Things done

Add 2 new python packages, which are new dependencies required by the new version of streamcontroller. `streamcontroller_streamdeck` is a fork of existing `streamdeck` package made by author of streamcontroller, required by the new version of streamcontroller.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sifmelcara commented:
Diff: https://github.com/StreamController/StreamController/compare/1.5.0-beta.8...1.5.0-beta.11
Release notes: https://github.com/StreamController/StreamController/releases

keenanweaver reviewed (approved):
Binding and actions work as expected.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 416567`
Commit: `73ba03b817c6d60bf6106dbbdd669c811b966583`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>python312Packages.python-wayland-extra</li>
    <li>python312Packages.python-wayland-extra.dist</li>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.python-wayland-extra</li>
    <li>python313Packages.python-wayland-extra.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
    <li>streamcontroller</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5589

keenanweaver commented:
Looks like `streamcontroller` isn't building now.

## `nixpkgs-review` result for [#416567](https://github.com/NixOS/nixpkgs/pull/416567)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 416567`

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/16279442997


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>streamcontroller</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.python-wayland-extra</li>
    <li>python312Packages.python-wayland-extra.dist</li>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.python-wayland-extra</li>
    <li>python313Packages.python-wayland-extra.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>streamcontroller</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.python-wayland-extra</li>
    <li>python312Packages.python-wayland-extra.dist</li>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.python-wayland-extra</li>
    <li>python313Packages.python-wayland-extra.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
  </ul>
</details>

andi242 commented:
the breaking of `textual` also broke `streamcontroller 1.5.0-beta.8` so we might really need this update to .11 with a fixed version textual. 
unfortunately the unbreaking of the textual seems to be stale https://github.com/NixOS/nixpkgs/pull/425707 and just reverting tree-sitter packages to an earlier version, which would not be enough for streamcontroller, right?

beta.11 is not available, while beta.8 is preventing system builds and updating all other packages that need updates, so a bit of a lock up at least for my setup.
Switched to flatpak instead, which is a bummer, but at least it works.

keenanweaver reviewed (approved):
All functionality seemingly working, at least from what I've tested on x86_64-linux.

## `nixpkgs-review` result for [#416567](https://github.com/NixOS/nixpkgs/pull/416567)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 416567`

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/16544533428


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>python312Packages.python-wayland-extra</li>
    <li>python312Packages.python-wayland-extra.dist</li>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.python-wayland-extra</li>
    <li>python313Packages.python-wayland-extra.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
    <li>streamcontroller</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>python312Packages.python-wayland-extra</li>
    <li>python312Packages.python-wayland-extra.dist</li>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.python-wayland-extra</li>
    <li>python313Packages.python-wayland-extra.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
    <li>streamcontroller</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.streamcontroller-streamdeck</li>
    <li>python312Packages.streamcontroller-streamdeck.dist</li>
    <li>python313Packages.streamcontroller-streamdeck</li>
    <li>python313Packages.streamcontroller-streamdeck.dist</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5727

SuperSandro2000 commented on pkgs/development/python-modules/python-wayland-extra/default.nix:
```diff
@@ -0,0 +1,43 @@
+    lxml
+    pytest
+    requests
+    ruff
```
```suggestion
```
that, too

sifmelcara commented on pkgs/development/python-modules/python-wayland-extra/default.nix:
```diff
@@ -0,0 +1,43 @@
+  build-system = [ hatchling ];
+
+  dependencies = [
+    black
```
If I remove any of `black`, `ruff`, or `pytest` from the dependencies, it doesn't build, as the [upstream `pyproject.toml`](https://github.com/dennisrijsdijk/python-wayland-extra/blob/main/pyproject.toml) lists them as dependencies. I moved them to `nativeCheckInputs` instead.

I'm not familiar with python ecosystem so let me know if I'm doing it wrong

SuperSandro2000 commented on pkgs/development/python-modules/streamcontroller-streamdeck/default.nix:
```diff
@@ -0,0 +1,43 @@
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
```
```suggestion
  pname = "streamcontroller-streamdeck";
  version = "0.1.5";
  pyproject = true;

  src = fetchPypi {
    inherit version;
    pname = "streamcontroller_streamdeck";
```

SuperSandro2000 commented on pkgs/development/python-modules/python-wayland-extra/default.nix:
```diff
@@ -0,0 +1,46 @@
+  src = fetchPypi {
+    inherit version pname;
+    hash = "sha256-HSBOCWP3o/BHmg3LO+LU+GpYkEYSqdljjYcEPdOnxZk=";
+  };
```
```suggestion
  };

  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace-fail ', "black", "lxml", "requests", "pytest", "ruff"' ""
  '';
```

sifmelcara committed:
streamcontroller: 1.5.0-beta.8 -> 1.5.0-beta.12

sifmelcara committed:
python3Packages.streamcontroller-streamdeck: init at 0.1.5

It's a fork of streamdeck package made by streamcontroller author,
required by streamcontroller.

pkgs/development/python-modules/streamcontroller-streamdeck/default.nix
is copied from pkgs/development/python-modules/streamdeck/default.nix
with minor modification.

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

sifmelcara committed:
python3Packages.python-wayland-extra: init at 0.7.0

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
